libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.navigation.core.routerefresh
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import kotlin.math.min
+
+internal class DirectionsRouteDiffProvider {
+
+    fun buildRouteDiffs(
+        oldRoute: DirectionsRoute,
+        newRoute: DirectionsRoute,
+        currentLegIndex: Int,
+    ): List<String> {
+        val oldRouteLegs = oldRoute.legs() ?: return emptyList()
+        val newRouteLegs = newRoute.legs() ?: return emptyList()
+        val routeDiffs = arrayListOf<String>()
+        for (legIndex in currentLegIndex until min(oldRouteLegs.size, newRouteLegs.size)) {
+            val oldLegAnnotation = oldRouteLegs[legIndex].annotation()
+            val newLegAnnotation = newRouteLegs[legIndex].annotation()
+            val updatedAnnotations = getUpdatedAnnotations(oldLegAnnotation, newLegAnnotation)
+            if (updatedAnnotations.isNotEmpty()) {
+                routeDiffs.add("Updated ${updatedAnnotations.joinToString()} at leg $legIndex")
+            }
+        }
+        return routeDiffs
+    }
+
+    private fun getUpdatedAnnotations(
+        oldLegAnnotation: LegAnnotation?,
+        newLegAnnotation: LegAnnotation?,
+    ): List<String> {
```

Comment:
nit:
maybe store in lexical order? or return map

Reply:
I used the order in which corresponding methods are located in `LegAnnotation` class. I think it doesn't really matter.